plugins/gnupgplugin/gpgme_p.h

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (C) 2018 Damien Caliste.
+ * Contact: Damien Caliste <dcaliste@free.fr>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#ifndef _FILE_OFFSET_BITS
+#define _FILE_OFFSET_BITS 64
+#endif
+#include <gpgme.h>
+
+#include <Crypto/key.h>
+#include <QString>
+#include <QDebug>
+#include <QDateTime>
+
+struct GPGmeContext {
+    gpgme_ctx_t ctx;
+    gpgme_error_t err;
+    GPGmeContext(gpgme_protocol_t protocol, const QString &home = QString())
+        : ctx(0), err(0)
+    {
+        err = gpgme_engine_check_version(protocol);
+        if (gpgme_err_code(err) != GPG_ERR_NO_ERROR) {
+            gpgme_engine_info_t info;
+            gpgme_error_t err2;
+            err2 = gpgme_get_engine_info(&info);
+            if (!err2) {
+                while (info) {
+                    if (info->protocol == protocol) {
+                        qWarning() << "protocol:" << gpgme_get_protocol_name(info->protocol);
```

Comment:
instead of qWarning, best to use qCWarning() and define a plugin-specific logging category.  Hrm, or perhaps we need to add a generic secrets daemon plugin logging category, which all plugins should use...

Reply:
I've added a commit adding logging category for plugins and used it accordingly.